benchmarks/benchmarks/common.py

```diff
@@ -10,12 +10,243 @@
 import itertools
 import random
 
+from asv_runner.benchmarks.mark import SkipNotImplemented
+
 
 class Benchmark:
     """
     Base class with sensible options
     """
-    pass
+
+
+class XPBenchmark(Benchmark):
+    """
+    Base class for benchmarks that are run on multiple Array API backends
+    and devices. Supports multiple devices, jax.jit, and lazy/asynchronous
+    evaluation.
+
+    Basic usage
+    -----------
+    ::
+        def myfunc(x):
+            return x + 1
+
+        class MyFunc(XPBenchmark):
+            def setup(self, backend):
+                super().setup(backend, myfunc)
+                x = self.xp.arange(5)
+                self.x = self.synchronize(x)
+                if self.warmup:
+                    self.func(self.x)
+
+            def time_myfunc(self, backend):
+                self.func(self.x)
+
+    Adding parameters
+    -----------------
+    In the below example:
+    - We add a `size` asv parameter
+    - We add a `plus` function parameter which can't be traced by jax.jit
+
+    ::
+        def myfunc(x, plus=True):
+            return x + 1 if plus else x - 1
+
+        class MyFunc(XPBenchmark):
+            param_names = (*XPBenchmark.param_names, "size")
+            params = (*XPBenchmark.params, [5, 10])
+
+            def setup(self, backend, size):
+                super().setup(backend, myfunc, static_argnames=("plus",))
+                x = self.xp.arange(size)
+                self.x = self.synchronize(x)
+                if self.warmup:
```

Comment:
One _could_ write some automatic discovery system that applies `jax.jit` automatically to all `time_` methods - as it's safe to assume that all asv parameters end up as static jax.jit parameters.
The risk of overengineering, or encroaching in the same type of magic system that jax definitely uses, is very high though.